### PR TITLE
Update Windows metadata

### PR DESCRIPTION
**Pull request by riverar**
Updates Windows.Win32, Windows.Wdk, and Windows metadata.

### CODE REVIEW

File: crates/targets/baseline/icu.dll.c
Comment (riverar) [inline crates/targets/baseline/icu.dll.c:1]: Just a heads up, breaking lib change here as a result of ICU API remapping for down-level support. https://github.com/microsoft/win32metadata/pull/1958.

**Comment by riverar** [comment]
This is ready for your review @kennykerr, no rush.

File: crates/libs/bindgen/src/metadata.rs [hunk 1]
```diff
@@ -240,7 +240,7 @@ impl Signature {
         }
 
         // Then we see if we can infer retval-like behavior more conservatively.
-        self.params.last().map_or(false, |param| param.is_retval())
+        self.params.last().is_some_and(|param| param.is_retval())
```
Comment (kennykerr) [inline crates/libs/bindgen/src/metadata.rs:243]: Clippy?

File: crates/targets/baseline/icu.dll.c
Comment (kennykerr) [reply crates/targets/baseline/icu.dll.c:1]: Actually, we should see if there are actually any omitted APIs - if they've just moved to a more compatible library then that may not be a breaking change. So long as none have been removed.

File: crates/libs/bindgen/src/metadata.rs (hunk 1 above)
Comment (riverar) [reply crates/libs/bindgen/src/metadata.rs:243]: Yep, can pull this out if you'd prefer to address in a standalone PR.

**Pull request merged**